Drone/drone_base.py

```diff
@@ -0,0 +1,361 @@
+# Standard Library
+import abc
+import coloredlogs
+import dronekit
+from dronekit import VehicleMode
+import logging
+import math
+import time
+import threading
+
+# Ours
+from ..Utilities import constants as c
+from ..Utilities import dronekit_wrappers as dkw
+
+class DroneBase(object):
+    """
+    Wraps a DroneKit.vehicle and sensors
+
+    Data Members
+    ----------
+    vehicle: DroneKit.Vehicle
+        Interface to see parameters and send commands
+    devices: List of Device (Device not yet implemented)
+        Interface to the drone's devices, such as camera
+    connected: Boolean
+        True if the drone is connected to Ardupilot
+    """
+    __metaclass__ = abc.ABCMeta
+
+    cnt = 0
+
+    def __init__(self):
+        self.vehicle = None
+        self.devices = []
+        self.connected = False
+        self.logger = logging.getLogger(__name__)
+
+    def connect(self, isInSimulator):
+        """
+        Attempts to connect to Ardupilot
+
+        Parameters
+        ----------
+        isInSimulator: Boolean
+            True if the program is intended to run in the Gazebo simulator
+            and False if the program is being run on the real-life drone
+
+        Precondition:
+        ----------
+        None
+
+        Postcondition:
+        ----------
+        Upon a successful connection, self.connected is set to True.
+        The drone is now suitable to be armed.
+
+        Returns:
+        ----------
+        None
+        """
+        if isInSimulator:
+            self.vehicle = dronekit.connect(c.CONNECTION_STRING_SIMULATOR, wait_ready=True, 
+                status_printer=None)
+            self.logger.info(threading.current_thread().name + ": Connecting to the simulated ardupilot")
+        else:
+            self.vehicle = dronekit.connect(c.CONNECTION_STRING_REAL, wait_ready=True, 
+                status_printer=None)
+            self.logger.info(threading.current_thread().name + ": Connecting to the real-life ardupilot")
+
+        if self.vehicle is not None:
+            self.connected = True
+
+    def altitude(self):
+        """
+        Simple wrapper function to retrieve the distance being read by 
+        the drone's rangefinder
+
+        Returns:
+        ----------
+        double
+            The distance from the ground
+        """
+        return self.vehicle.rangefinder.distance
+
+    def altitude_failsafe(self):
+        """
+        Simple wrapper function to retrieve the distance being read by 
+        the drone's barometer
+
+        Returns:
+        ----------
+        double
+            The distance from the ground
+        """
+        return self.vehicle.location.global_relative_frame.alt
+
+    def arm(self, timeout = 60):
```

Comment:
Consider creating a constant for the arm timeout like so:
```suggestion
    def arm(self, timeout = c.DEFAULT_ARM_TIMEOUT):
```

Reply:
Also, consider adding a mode parameter to arm the drone in a different mode (good for if we want to add our own flight mode or if we just decide to use a different flight mode)
```suggestion
    def arm(self, timeout = c.DEFAULT_ARM_TIMEOUT, mode = c.GUIDED):
```